WireAPI/Sources/WireAPI/APIs/ClientAPI/ClientAPIV0.swift

```diff
@@ -0,0 +1,175 @@
+        let request = HTTPRequest(
+            path: "/users/list-clients/v2", /// v2 suffix required for api version v0 and v1, suffix removed from next versions
+            method: .post,
+            body: body
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: .ok, type: UserClientsV0.self)
+            .parse(response)
+    }
+}
+
+struct ListUserClientV0: Decodable, ToAPIModelConvertible {
+
+    let payload: [UserClientV0]
+
+    init(from decoder: any Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let payload = try container.decode([UserClientV0].self)
+        self.payload = payload
+    }
+
+    func toAPIModel() -> [UserClient] {
+        payload.map { $0.toAPIModel() }
+    }
+}
+
+struct UserClientV0: Decodable, ToAPIModelConvertible {
+
+    let id: String
+    let type: UserClientType
+    let activationDate: UTCTimeMillis
+    let label: String?
+    let model: String?
+    let deviceClass: DeviceClass?
+    let lastActiveDate: UTCTime?
+    let mlsPublicKeys: MLSPublicKeys?
+    let cookie: String?
+    let capabilities: CapabilitiesList?
+
+    struct CapabilitiesList: Decodable {
+
+        let capabilities: [UserClientCapability]
+
+    }
+
+    enum CodingKeys: String, CodingKey {
+
+        case id
+        case type
+        case activationDate = "time"
+        case label
+        case model
+        case deviceClass = "class"
+        case lastActiveDate = "last_active"
+        case mlsPublicKeys = "mls_public_keys"
+        case cookie
+        case capabilities
+
+    }
+
+    func toAPIModel() -> UserClient {
+        UserClient(
+            id: id,
+            type: type,
+            activationDate: activationDate.date,
+            label: label,
+            model: model,
+            deviceClass: deviceClass,
+            lastActiveDate: lastActiveDate?.date,
+            mlsPublicKeys: mlsPublicKeys,
+            cookie: cookie,
+            capabilities: capabilities?.capabilities ?? []
+        )
+    }
+
+}
+
+struct UserClientsRequestV0: Encodable {
+
+    let qualifiedIDs: [UserID]
+
+    enum CodingKeys: String, CodingKey {
+        case qualifiedIDs = "qualified_users"
+    }
+
+}
+
+struct UserClientsV0: Decodable, ToAPIModelConvertible {
+    typealias Domain = String
+    typealias UserID = String
+
+    let payload: [Domain: [UserID: [SimplifiedUserClient]]]
+
+    enum CodingKeys: String, CodingKey {
+        case qualifiedUserMap = "qualified_user_map"
+    }
+
+    init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let payload = try container.decode(
+            [Domain: [UserID: [SimplifiedUserClient]]].self,
+            forKey: .qualifiedUserMap
+        )
+
+        self.payload = payload
+    }
```

Comment:
Again is this necessary? Can we achieve this with a type alias?

```swift
typealias UserClientsV0 = [Domain: UserID: [SimplifiedUserClient]]]`
```

Reply:
I simplified it a little bit in https://github.com/wireapp/wire-ios/pull/2019/commits/b9f535f64fb834f6a0409f69830e48d7c82c8f45